lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -171,8 +172,11 @@ monitorStakePools tr (block0, Quantity k) nl db@DBLayer{..} = do
     cursor <- initCursor
     traceWith tr $ MsgStartMonitoring cursor
     follow nl trFollow cursor forward header >>= \case
-        Nothing    -> pure ()
-        Just point -> do
+        FollowInterrupted -> pure ()
+        FollowFatal -> pure ()
+        FollowException ->
+            monitorStakePools tr (block0, Quantity k) nl db
```

Comment:
I take it this is _the_ fix?

This couldn't lead to an infinite loop of exceptions? E.g. if something is wrong with the DB?

Reply:
It could. It may be better to actually add a little pause here. So that we don't end up in a loop retrying too fast too often.